File: 03.programmers/lv02/02.Python/Matrix_mul.py
def solution(arr1, arr2):
    return [[sum(a*b for a, b in zip(arr1_row,arr2_col)) for arr2_col in zip(*arr2)] for arr1_row in arr1]

input_arr1 = [[1, 4], [3, 2], [4, 1]]
input_arr2 = [[3, 3], [3, 3]]

print(solution(input_arr1,input_arr2))

File: 03.programmers/lv02/01.C++/Matrix_mul.cpp
#include<iostream>
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer;
    for (int i = 0; i < arr1.size(); i++) {
		vector<int> tmp;
		for (int j = 0; j < arr2[0].size(); j++) {
			int sum = 0;
			for (int k = 0; k < arr1[0].size(); k++) {
				sum += arr1[i][k] * arr2[k][j];
			}
			tmp.push_back(sum);
		}
		answer.push_back(tmp);
	}
    return answer;
}
int main(){
    vector<vector<int>> arr1 = {{1, 4},{3, 2}, {4, 1}};
    vector<vector<int>> arr2 = {{3,3},{3,3}};

    for(auto i : solution(arr1, arr2))
        for(auto j : i)
            cout<<j<<" ";

    return 0;
}